core/appeal/service.go

```diff
@@ -486,6 +493,367 @@ func validateAppealOnBehalf(a *domain.Appeal, policy *domain.Policy) error {
+		if appeal.Resource == nil {
+			resource, err := s.resourceService.Get(egctx, &domain.ResourceIdentifier{ID: appeal.ResourceID})
+			if err != nil {
+				return fmt.Errorf("error getting resource: %w", err)
+			}
+			appeal.Resource = resource
+		}
+		return nil
+	})
+
+	eg.Go(func() error {
+		providersData, err := s.getProvidersMap(egctx)
+		if err != nil {
+			return fmt.Errorf("error getting providers map: %w", err)
+		}
+		providers = providersData
+		return nil
+	})
+
+	eg.Go(func() error {
+		policiesData, err := s.getPoliciesMap(egctx)
+		if err != nil {
+			return fmt.Errorf("error getting policies map: %w", err)
+		}
+		policies = policiesData
+		return nil
+	})
+
+	eg.Go(func() error {
+		pendingAppealsData, err := s.getAppealsMap(egctx, &domain.ListAppealsFilter{
+			Statuses:   []string{domain.AppealStatusPending},
+			AccountIDs: []string{appeal.AccountID},
+		})
+		if err != nil {
+			return fmt.Errorf("error while listing pending appeals: %w", err)
+		}
+		pendingAppeals = pendingAppealsData
+		return nil
+	})
+
+	if err := eg.Wait(); err != nil {
+		return err
+	}
+
+	appeal.SetDefaults()
+
+	if appeal.AccountID != existingAppeal.AccountID || appeal.ResourceID != existingAppeal.ResourceID || appeal.Role != existingAppeal.Role {
+		if err := validateAppeal(appeal, pendingAppeals); err != nil {
+			return err
+		}
+	}
+
+	provider, err := getProvider(appeal, providers)
+	if err != nil {
+		return err
+	}
+
+	var policy *domain.Policy
+	if isAdditionalAppealCreation && appeal.PolicyID != "" && appeal.PolicyVersion != 0 {
+		policy = policies[appeal.PolicyID][appeal.PolicyVersion]
+	} else {
+		policy, err = getPolicy(appeal, provider, policies)
+		if err != nil {
+			return err
+		}
+	}
+
+	activeGrant, err := s.findActiveGrant(ctx, appeal)
+	if err != nil && err != ErrGrantNotFound {
+		return err
+	}
+
+	if activeGrant != nil {
+		if err := s.checkExtensionEligibility(appeal, provider, policy, activeGrant); err != nil {
+			return err
+		}
+	}
+
+	if err := s.providerService.ValidateAppeal(ctx, appeal, provider, policy); err != nil {
+		return fmt.Errorf("provider validation: %w", err)
+	}
+
+	strPermissions, err := s.getPermissions(ctx, provider.Config, appeal.Resource.Type, appeal.Role)
+	if err != nil {
+		return fmt.Errorf("getting permissions list: %w", err)
+	}
+	appeal.Permissions = strPermissions
+
+	if err := validateAppealDurationConfig(appeal, policy); err != nil {
+		return err
+	}
+
+	if err := validateAppealOnBehalf(appeal, policy); err != nil {
+		return err
+	}
+
+	if err := s.populateAppealMetadata(ctx, appeal, policy); err != nil {
+		return fmt.Errorf("getting appeal metadata: %w", err)
+	}
+
+	if err := s.addCreatorDetails(ctx, appeal, policy); err != nil {
+		return fmt.Errorf("getting creator details: %w", err)
+	}
+
+	// mark previous approvals as stale
+	for _, approval := range existingAppeal.Approvals {
+		approval.IsStale = true
+		err := s.approvalService.UpdateApproval(ctx, approval)
```

Comment:
we can skip doing this by marking each previous approvals as stale and then append them in `appeal.Approvals[]`, `appealRepository.Update` should handle the updating the old approvals and inserting new approvals